Sources/BraveWallet/NetworkIcon.swift

```diff
@@ -79,6 +79,10 @@ struct NetworkIcon: View {
       return ("optimism", false)
     case BraveWallet.AuroraMainnetChainId:
       return ("aurora", false)
+    case BraveWallet.FilecoinEthereumMainnetChainId:
+      return ("filecoin-asset-icon", false)
+    case BraveWallet.FilecoinEthereumTestnetChainId:
+      return ("filecoin-asset-icon", true)
     default:
       return nil
```

Comment:
Similar to above comment about duplicated code, I think we can make use of an enum to get these string values (perhaps with function accepting a `chainId`), and I _think_ we could use a simple check against `WalletConstants.supportedTestNetworkChainIds` to see if we should grayscale the image or not:
```let isGrayscale = WalletConstants.supportedTestNetworkChainIds.contains(network.chainId)```

Reply:
updated in [188d673](https://github.com/brave/brave-ios/pull/7361/commits/188d6732e73af0f5cf9271d747aef6b7aedb266b)

Reply:
Can we use that `isGrayscale` flag to reduce code here?